tests/test_frompyobject.rs

```diff
@@ -0,0 +1,308 @@
+use pyo3::exceptions::PyValueError;
+use pyo3::prelude::*;
+use pyo3::types::{PyDict, PyString, PyTuple};
+use pyo3::{PyErrValue, PyMappingProtocol};
+
+#[macro_use]
+mod common;
+
+#[derive(Debug, FromPyObject)]
+pub struct A<'a> {
+    #[extract(getattr)]
+    s: String,
+    #[extract(get_item)]
+    t: &'a PyString,
+    #[extract(getattr("foo"))]
```

Comment:
The nice thing about the pyo3 form is that it could compose

```suggestion
    #[pyo3(attribute, name = "foo")]
```